tools/clang/test/CodeGenHLSL/selectObj4.hlsl

```diff
@@ -0,0 +1,25 @@
+// RUN: %dxc -E main -T cs_6_0 %s | FileCheck %s
+
+// CHECK: select
+// CHECK: i32 2, i32 1
+// CHECK: select
+// CHECK: i32 0, i32 3
+
+
+RWStructuredBuffer<float2x2> o[6];
+
+StructuredBuffer<float2x2> i[6];
+
+uint s;
+
+void run(uint id) {
+RWStructuredBuffer<float2x2> ot = s > 8 ? o[2]:o[1];
```

Comment:
Can you add some tests for the error cases you have in your code where you fail to simplify the select operands (for ResouceClass and ResourceID)